bookList-no-cell/components/serch-books/search-books.js

```diff
@@ -0,0 +1,60 @@
+import { LitElement, html } from "lit";
+
+import search from "./search-books.styles";
+
+export class searchBooks extends LitElement {
+  static properties = {
+    data: { type: Array },
+    dataFilter: { type: Array },
+  };
+
+  constructor() {
+    super();
+    this.word = "";
+    this.data = [{}];
+    this.dataFilter = [{}];
+  }
+
+  static get styles() {
+    return [search];
+  }
+
+  filterBook(e) {
+    this.dataFilter = [];
+    let wordFilter = e.target?.value.toLowerCase();
+
+    this.data?.map((book) => {
+      const title = book.title.toLowerCase();
+      const author = book.author.toLowerCase();
+
+      if (
+        title.indexOf(wordFilter) !== -1 ||
+        author.indexOf(wordFilter) !== -1
+      ) {
+        this.dataFilter = [...this.dataFilter, book];
+      }
+    });
```

Comment:
instead of using the map function you can use the `filter` function and assign it directly to the variable